assets/react/controllers/Transporter/TransporterTimeline.jsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from "react";
+import { Timeline, Image } from "antd";
+import { useTranslation } from "react-i18next";
+import {
+  DownloadOutlined,
+  LoadingOutlined,
+  UploadOutlined,
+  QuestionCircleOutlined,
+} from "@ant-design/icons";
+
+function ediDot(edi) {
+  const style = { fontSize: "20px" };
+  switch (edi.direction) {
+    case "INBOUND":
+      return <DownloadOutlined style={style} />;
+    case "OUTBOUND":
+      switch (edi.syncedAt) {
+        case null:
+          return <LoadingOutlined style={style} />;
+        default:
+          return <UploadOutlined style={style} />;
+      }
+    default:
+      return <QuestionCircleOutlined style={style} />;
+  }
+}
+
+function ediColor(edi) {
+  if (edi.messageType === "SCONTR") {
+    return "blue";
+  }
+
+  switch (edi?.subMessageType?.split("|")[0]) {
```

Comment:
I'd suggest making it a function like `getSituation` to make it easier to understand. And re-use it on line 47